flytekit/types/directory/types.py

```diff
@@ -322,6 +323,10 @@ def to_literal(
 
         remote_directory = None
         should_upload = True
+        batch_size = get_batch_size(python_type)
+
+        if is_annotated(python_type):
```

Comment:
what is this?

Reply:
this is just we can call `get_format` and still get the actual extension of the FlyteDirectory if it's set. We could move this to `get_format`, right? Also, we should probably add a couple of unit tests to verify this.

Reply:
removed it

Reply:
but like... shouldn't this be handled at the type engine layer?

Reply:
I remove the code that checks if it's annotated. I just found that it still works if FlyteDirecrory is annotated. 
```python
flyte_dir = Annotated[FlyteDirectory, BatchSize(100)]
print(flyte_dir.extension()0 # it still works
```